src/extension/migration.js

```diff
@@ -0,0 +1,32 @@
+export default function migrateStyle(properties) {
+  if (properties.components && properties.components.length > 0) return properties;
+
+  const dataTemplateLabel = {
+    key: "label",
+    label: {
+      value: {
+        ...properties.style?.fontColor,
+      },
+    },
+  };
+
+  const dataTemplateBackground = {
+    key: "backgroundColor",
+    backgroundColor: {
+      ...properties.style?.backgroundColor,
+    },
+  };
+
+  const dataTemplateBorder = {
+    key: "border",
+    border: {
+      ...properties.style?.border,
+    },
+  };
+
+  if (!properties.components) properties.components = [];
```

Comment:
It looks much better now!

But this can still be improved a little by removing the if statement and assigning components array directly: `properties.components = [dataTemplateLabel, dataTemplateBackground, dataTemplateBorder];`